include/kernel.h

```diff
@@ -506,9 +506,16 @@ struct _thread_stack_info {
 
 	/* Stack Size - Thread writable stack buffer size. Represents
 	 * the size of the actual area, starting from the start member,
-	 * that should be writable by the thread
+	 * that should be writable by the thread. This includes any TLS storage
+	 * and random offsets applied to the initial stack pointer.
 	 */
 	size_t size;
+
+	/* A positive adjustment value to the size member, removing any storage
```

Comment:
Change 'positive' to 'non-negative', since this in general may be zero.

Reply:
Suggestion:
`Non-negative adjustment value to the thread's writable stack buffer, reflecting any area reserved for local thread data storage, or left-out due to random adjustments applied to the initial thread stack pointer during thread initialization. (start + size - offset) is the initial stack pointer of a thread.`

Reply:
size_t is unsigned so I probably don't need to mention negative numbers at all :) I'll revise.